consensus/ibft/src/main/java/tech/pegasys/pantheon/consensus/ibft/validation/RoundChangeMessageValidator.java

```diff
@@ -0,0 +1,152 @@
+
+      return validatePrepareCertificate(certificate);
+    }
+
+    return true;
+  }
+
+  private boolean validateRoundChangeIsForCurrentHeightAndTargetsFutureRound(
+      final IbftSignedMessageData<IbftUnsignedRoundChangeMessageData> msg) {
+
+    final ConsensusRoundIdentifier msgRoundIdentifier =
+        msg.getUnsignedMessageData().getRoundChangeIdentifier();
+
+    if (msgRoundIdentifier.getSequenceNumber() != currentRound.getSequenceNumber()) {
+      LOG.info("Invalid RoundChange message, not valid for local chain height.");
+      return false;
+    }
+
+    if (msgRoundIdentifier.getRoundNumber() <= currentRound.getRoundNumber()) {
+      LOG.info("Invalid RoundChange message, targets a round which is not in the future");
+      return false;
+    }
+
+    return true;
+  }
+
+  private boolean validatePrepareCertificate(final IbftPreparedCertificate certificate) {
+    final IbftSignedMessageData<IbftUnsignedPrePrepareMessageData> preprepareMessage =
+        certificate.getIbftPrePrepareMessage();
+
+    final ConsensusRoundIdentifier prepareCertRound =
+        preprepareMessage.getUnsignedMessageData().getRoundIdentifier();
+
+    if (!validatePreprepareCertificateRound(prepareCertRound)) {
+      return false;
+    }
+
+    final MessageValidator messageValidator = messageValidatorFactory.createAt(prepareCertRound);
+    return validateConsistencyOfPrepareCertificateMessages(certificate, messageValidator);
+  }
+
+  private boolean validateConsistencyOfPrepareCertificateMessages(
+      final IbftPreparedCertificate certificate, final MessageValidator messageValidator) {
+
+    if (!messageValidator.addPreprepareMessage(certificate.getIbftPrePrepareMessage())) {
+      LOG.info("Invalid RoundChange message, embedded Preprepare message failed validation.");
+      return false;
+    }
+
+    if (certificate.getIbftPrepareMessages().size() < minimumPrepareMessages) {
+      LOG.info(
+          "Invalid RoundChange message, insufficient prepare messages exist to justify "
+              + "prepare certificate.");
+      return false;
+    }
+
+    for (final IbftSignedMessageData<IbftUnsignedPrepareMessageData> prepareMsg :
+        certificate.getIbftPrepareMessages()) {
+      if (!messageValidator.validatePrepareMessage(prepareMsg)) {
+        LOG.info("Invalid RoundChange message, embedded Prepare message failed validation.");
+        return false;
+      }
+    }
+
+    return true;
+  }
+
+  private boolean validatePreprepareCertificateRound(
+      final ConsensusRoundIdentifier prepareCertRound) {
+
+    if (prepareCertRound.getSequenceNumber() != currentRound.getSequenceNumber()) {
+      LOG.info("Invalid RoundChange message, PreprepareCertificate is not for local chain height.");
+      return false;
+    }
+
+    if (prepareCertRound.getRoundNumber() > currentRound.getRoundNumber()) {
+      LOG.info("Invalid RoundChange message, PreprepareCertificate is not for a past round.");
```

Comment:
The correct check is
 pepareCertRound.getRoundNumber() < RounChnageMesage.getRoundChangeIdentifier().getRoundIdentifier()